### ISSUE TITLE
Public API test fails in some browsers

### ISSUE THREAD

**Issue by mlaval** [angular/angular#5072]
It started with the last RxJS upgrade in "old" browsers.

In IE11, Safari <= 8 and Android <= 4.4:
```
Expected [ '+ Observable.@@observable()', '+ Subject.@@observable()' ] to equal [  ].
```

In IE9 and IE10:
```
Expected [ '+ Observable.@@observable()', '- Subject#combineLatest():js', '- Subject#concat():js',
'- Subject#defer():js', '- Subject#empty():js', '- Subject#forkJoin():js', 
'- Subject#from():js', '- Subject#fromArray():js', '- Subject#fromEvent():js', 
'- Subject#fromEventPattern():js', '- Subject#fromPromise():js', '- Subject#interval():js', 
'- Subject#merge():js', '- Subject#never():js', '- Subject#of():js', '- Subject#range():js', 
'- Subject#throw():js', '- Subject#timer():js', '- Subject#zip():js', '+ Subject.@@observable()' ] 
to equal [  ].
```

**Comment by ocombe** [comment]
Well if you look at the sauce lab results you can see that on the main page of the repo:

![Sauce Lab Results](https://camo.githubusercontent.com/efa5a8b6df23b248c39a72d85bc654db9a10def5/68747470733a2f2f73617563656c6162732e636f6d2f62726f777365722d6d61747269782f616e67756c6172322d63692e737667)

**Comment by mlaval** [comment]
Some findings about the 2 problems:

1) `@@observable` is the token used in RxJS to define `Symbol.observable` when the browser doesn't support `Symbol.for()`, see https://github.com/ReactiveX/RxJS/blob/master/src/util/Symbol_observable.ts
In the `Observable` class, it is then exported to match the spec https://github.com/ReactiveX/RxJS/blob/master/src/Observable.ts#L64-L71
-> suggestion: ignore this inconsistent API when checking, sent PR #5093 for that

2) The `Subject#` entries which are missing in IE9 and IE10 are the static methods of the parent class `Observable`: https://github.com/ReactiveX/RxJS/blob/master/src/Observable.ts#L136-L153
For some reason, they are not correctly inherited in these browsers.

**Comment by mlaval** [comment]
The second problem happens because IE9 and IE10 do not support [Object.setPrototypeOf](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Object/setPrototypeOf) which is used by Babel for [class inheritance](https://github.com/babel/babel/blob/8ab4a5df433accf65a127dfe357dae8ffd6f498b/packages/babel/src/transformation/templates/helper-inherits.js#L13).
RxJS uses Babel to transpile from es6 to cjs, that we use to run the unit tests.
It seems to be a known limitation: https://github.com/babel/babel/issues/2450

**Comment by robwormald** [comment]
FYI changed Rx to fully transpile with TS to CJS, so that will probably solve any Babel specific behavior.

**Comment by robwormald** [comment]
@mlaval looks like `Subject` has been removed from the public API, so that solves most of this problem, I guess? The Symbol issue is yet to to be resolved though.

**Comment by mlaval** [comment]
Correct, we're only left with this error in IE11, Safari <= 8 and Android <= 4.4:
`Expected [ '+ Observable.@@observable()' ] to equal [  ].`

Sent https://github.com/angular/angular/pull/5093 to avoid it, not sure if we can get a cleaner fix

**Comment by robwormald** [comment]
closed as #5093 fixes
